### DISCUSSION TITLE
GSoC: Working on "Reading Cross-Platform Locale Settings in Rust"

### DISCUSSION THREAD

**Discussion by JMoogs** [unicode-org/icu4x#4729] [General]
Hello all,

I'm interested in working on the "Reading Cross-Platform Locale Settings in Rust" project for GSoC this year. I've been researching into the task, and have proposed some possible methods in the relevant issue (3990). I believe I'm well suited to this project, and that I'd be able to complete all required tasks. 

I'm hoping I can be given some guidance on where to begin with the project and the logistics on where my code should be placed. I'd also like to ask which additional operating systems should be supported if possible. Additionally, I would appreciate if someone could point me towards some related issues that I could begin contributing to now, as this project seems quite detached from the rest of the crate.

Thanks!

**Comment by ashu26jha** [comment]
Hello @JMoogs I think this code should land in `utils` as mentioned by @sffc. It also makes sense that this crate will be used by devs only.

PS:

I am also a contributor and looking to tackle this in GSoC

**Comment by yolade** [comment]
Likewise @JMoogs  @ashu26jha I think we can work together in this crate.

**Reply by yolade** [reply]
@sffc Please guide us on where to start.

**Reply by ashu26jha** [reply]
Hello @yolade If there is something you need help with given that I know about it. I'd be happy to help. 

I think the best place start looking should be #3990. Here issue is described along with the requirements. Also both us has proposed some solutions, may have a look.

**Reply by yolade** [reply]
Tks @ashu26jha 
I'm taking a look at it.
